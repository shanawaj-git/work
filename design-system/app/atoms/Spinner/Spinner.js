import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const Spinner = props => {
  const { variant } = props;
  return (
    <img
      className="spinner-animation"
      alt=""
      src={getSpinnerVariantName(variant)}
    />
  );
};

export const getSpinnerVariantName = variant =>
  variant === 'light'
    ? require('./spinner_light.svg')
    : require('./spinner_dark.svg');

Spinner.propTypes = {
  variant: PropTypes.string,
};

export default Spinner;
