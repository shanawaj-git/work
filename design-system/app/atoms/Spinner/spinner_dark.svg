<svg id="SvgjsSvg1048" width="288" height="288" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs"><defs id="SvgjsDefs1049"></defs><g id="SvgjsG1050"><svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 512 512" viewBox="0 0 512 512" width="288" height="288"><path d="M256,150.6c-9,0-15.1-6-15.1-15.1V15.1C240.9,6,247,0,256,0c9,0,15.1,6,15.1,15.1v120.5C271.1,144.6,265,150.6,256,150.6z" fill="#faf9f9" class="color000 svgShape"></path><path fill="#e2e2e2" d="M256 361.4c-9 0-15.1 6-15.1 15.1v120.5c0 9 6 15.1 15.1 15.1 9 0 15.1-6 15.1-15.1V376.5C271.1 367.4 265 361.4 256 361.4M376.6 34c-4.8 0-9.1 2.7-12.2 6.7l-60.2 103.9c-4.5 7.5-1.5 15.1 4.5 19.6 3 1.5 4.5 1.5 7.5 1.5 4.5 0 9-3 12-6l60.2-103.9c4.5-7.5 1.5-15.1-4.5-19.6C381.5 34.6 379 34 376.6 34M195.9 347.2c-4.8 0-9.1 2.7-12.2 6.7l-60.2 103.9c-4.5 7.5-1.5 15.1 4.5 19.6 3 1.5 4.5 1.5 7.5 1.5 4.5 0 9-3 12-6l60.2-103.9c4.5-7.5 1.5-15.1-4.5-19.6C200.8 347.9 198.3 347.2 195.9 347.2" class="colorb2b2b2 svgShape"></path><path fill="#5c6067" d="M134.7,32.5c-2.4,0-4.7,0.7-6.7,2.2c-6,4.5-9,13.6-4.5,19.6l60.2,103.9c3,4.5,7.5,7.5,12,7.5c3,0,4.5,0,7.5-3
		c6-4.5,9-13.6,4.5-19.6L147.6,39.2C144.6,35.1,139.5,32.5,134.7,32.5" class="color121212 svgShape"></path><path fill="#e2e2e2" d="M315.5,345.7c-2.4,0-4.7,0.7-6.7,2.2c-6,4.5-9,13.6-4.5,19.6l60.2,103.9c3,4.5,7.5,7.5,12,7.5
		c3,0,4.5-1.5,7.5-3c6-4.5,9-13.6,4.5-19.6l-60.2-103.9C325.3,348.4,320.3,345.7,315.5,345.7" class="colorb2b2b2 svgShape"></path><path fill="#717479" d="M135.5,240.9H15.1C6,240.9,0,247,0,256c0,9,6,15.1,15.1,15.1h120.5c9,0,15.1-6,15.1-15.1
		C150.6,247,144.6,240.9,135.5,240.9" class="color595959 svgShape"></path><path fill="#e2e2e2" d="M496.9,240.9H376.5c-9,0-15.1,6-15.1,15.1c0,9,6,15.1,15.1,15.1h120.5c9,0,15.1-6,15.1-15.1
		C512,247,506,240.9,496.9,240.9" class="colorb2b2b2 svgShape"></path><path fill="#65686f" d="M48.3,121.3c-4.8,0-9.1,2.7-12.2,6.7c-4.5,7.5-1.5,15.1,4.5,19.6l103.9,60.2c3,1.5,4.5,1.5,7.5,1.5
		c4.5,0,9-1.5,12-6c4.5-7.5,1.5-15.1-4.5-19.6L55.7,123.5C53.2,122,50.7,121.3,48.3,121.3" class="color242424 svgShape"></path><path fill="#e2e2e2" d="M361.5,302c-4.8,0-9.1,2.7-12.2,6.7c-4.5,7.5-1.5,15.1,4.5,19.6l103.9,60.2c3,1.5,4.5,1.5,7.5,1.5
		c4.5,0,9-1.5,12.1-6c4.5-7.5,1.5-15.1-4.5-19.6l-103.9-60.2C366.4,302.7,363.9,302,361.5,302" class="colorb2b2b2 svgShape"></path><path fill="#7b7d81" d="M149.8,300.5c-2.4,0-4.7,0.7-6.7,2.2L39.2,362.9c-6,4.5-9,13.6-4.5,19.6c3,4.5,7.5,7.5,12.1,7.5c3,0,6,0,7.5-3
		l103.9-60.2c6-4.5,9-13.6,4.5-19.6C159.6,303.2,154.6,300.5,149.8,300.5" class="color8f8f8f svgShape"></path><path fill="#e2e2e2" d="M463,119.8c-2.4,0-4.7,0.7-6.8,2.2l-103.9,60.2c-6,4.5-9,13.6-4.5,19.6c3,4.5,7.5,7.5,12,7.5c3,0,4.5,0,7.5-3
		l103.9-60.2c6-4.5,9-13.6,4.5-19.6C472.8,122.5,467.8,119.8,463,119.8" class="colorb2b2b2 svgShape"></path></svg></g></svg>