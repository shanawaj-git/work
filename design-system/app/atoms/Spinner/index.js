import Spinner, { getSpinnerVariantName } from './Spinner';
export { getSpinnerVariantName };
export default Spinner;
