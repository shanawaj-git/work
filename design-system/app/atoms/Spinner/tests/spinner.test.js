import React from 'react';
import renderer from 'react-test-renderer';

import Spinner, { getSpinnerVariantName } from '../index';

describe('<Spinner />', () => {
  it('should render dark spinner', () => {
    const tree = renderer.create(<Spinner variant="contained" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render light spinner', () => {
    const tree = renderer.create(<Spinner variant="outlined" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render dark spinner based on variant prop', () => {
    expect(getSpinnerVariantName('dark')).toEqual('IMAGE_MOCK');
  });

  it('should render light spinner based on variant prop', () => {
    expect(getSpinnerVariantName('light')).toEqual('IMAGE_MOCK');
  });
});
