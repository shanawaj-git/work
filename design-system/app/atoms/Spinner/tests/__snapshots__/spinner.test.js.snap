// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render dark spinner 1`] = `
<img
  alt=""
  className="spinner-animation"
  src="IMAGE_MOCK"
/>
`;

exports[`<Spinner /> should render light spinner 1`] = `
<img
  alt=""
  className="spinner-animation"
  src="IMAGE_MOCK"
/>
`;
